app/src/main/java/com/keylesspalace/tusky/view/SliderPreference.kt

```diff
@@ -0,0 +1,185 @@
+        valueTo = a.getFloat(R.styleable.SliderPreference_android_valueTo, defaultValueTo)
+        stepSize = a.getFloat(R.styleable.SliderPreference_android_stepSize, defaultStepSize)
+        format = a.getString(R.styleable.SliderPreference_format) ?: defaultFormat
+
+        val decrementIconResource = a.getResourceId(R.styleable.SliderPreference_iconStart, -1)
+        if (decrementIconResource != -1) {
+            decrementIcon = AppCompatResources.getDrawable(context, decrementIconResource)
+        }
+
+        val incrementIconResource = a.getResourceId(R.styleable.SliderPreference_iconEnd, -1)
+        if (incrementIconResource != -1) {
+            incrementIcon = AppCompatResources.getDrawable(context, incrementIconResource)
+        }
+
+        a.recycle()
+    }
+
+    override fun onGetDefaultValue(a: TypedArray, i: Int): Any {
+        return a.getFloat(i, defaultValue)
+    }
+
+    override fun onSetInitialValue(defaultValue: Any?) {
+        value = getPersistedFloat((defaultValue ?: Companion.defaultValue) as Float)
+    }
+
+    override fun onBindViewHolder(holder: PreferenceViewHolder) {
+        super.onBindViewHolder(holder)
+        binding = PrefSliderBinding.bind(holder.itemView)
+
+        binding.root.isClickable = false
+
+        binding.slider.addOnChangeListener(this)
+        binding.slider.addOnSliderTouchListener(this)
+        binding.slider.value = value // sliderValue
+        binding.slider.valueTo = valueTo
+        binding.slider.valueFrom = valueFrom
+        binding.slider.stepSize = stepSize
+
+        // Disable the label, the value is shown in the preference summary
+        binding.slider.labelBehavior = LABEL_GONE
+        binding.slider.isEnabled = isEnabled
+
+        binding.summary.visibility = VISIBLE
+        binding.summary.text = formatter(value)
+
+        decrementIcon?.let { icon ->
+            binding.decrement.icon = icon
+            binding.decrement.visibility = VISIBLE
+            binding.decrement.setOnClickListener {
+                value -= stepSize
+            }
+        }
+
+        incrementIcon?.let { icon ->
+            binding.increment.icon = icon
+            binding.increment.visibility = VISIBLE
+            binding.increment.setOnClickListener {
+                value += stepSize
+            }
+        }
+    }
+
+    override fun onValueChange(slider: Slider, value: Float, fromUser: Boolean) {
+        if (!fromUser) return
+        binding.summary.text = formatter(value)
+    }
+
+    override fun onStartTrackingTouch(slider: Slider) {
+        // Deliberately empty
+    }
+
+    override fun onStopTrackingTouch(slider: Slider) {
+        value = slider.value
+    }
+
+    companion object {
+        const val TAG = "FloatSeekBarPreference"
+        const val defaultValueFrom = 0F
+        const val defaultValueTo = 1F
+        const val defaultValue = 0.5F
+        const val defaultStepSize = 0.1F
+        const val defaultFormat = "%3.1f"
```

Comment:
These should probably all be private. And `TAG` is both unused & has the wrong value.